### PR TITLE
Mes 2658 ensure candidate communications are correct for cat be

### PR DESCRIPTION
**Pull request by appleabs**
Mes-2658 Ensure candidate communications are correct for cat be. Created tests for cat-be and added competency labels for reverse and uncouple / recouple.

### CODE REVIEW

File: src/functions/sendCandidateResults/application/service/__tests__/personalisation-provider.cat-be.spec.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      const result = personalisationProvider.getEmailPersonalisation(completedCatBETest);
+
+      expect(result.drivingFaults.length).toBe(1);
+      expect(result.drivingFaults).toContain(`${welshCompetencyLabels.vehicleChecks}, 4`);
+
+      // expect(result.seriousFaults.length).toBe(2);
+      // expect(result.seriousFaults).toContain(`${welshCompetencyLabels.reverseLeftControl}`);
+
+      // expect(result.dangerousFaults.length).toBe(1);
+      // expect(result.dangerousFaults).toContain(`${welshCompetencyLabels.uncoupleRecouple}`);
```
Comment (rossfauldsbjss) [inline src/functions/sendCandidateResults/application/service/__tests__/personalisation-provider.cat-be.spec.ts:69]: If these are commented out for temporary reasons (e.g until welsh translations available) then they should be marked with a TODO comment. If not they should be removed.

File: src/functions/sendCandidateResults/application/service/__tests__/terminated-test.cat-be.spec.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+          outcomePayload: {
+            interface: NOTIFY_INTERFACE,
+            state: ProcessingStatus.ACCEPTED,
+            staff_number: '123456',
+            retry_count: 0,
+            error_message: null,
+          },
+        });
+        done();
+      },         1000);
```
Comment (rossfauldsbjss) [inline src/functions/sendCandidateResults/application/service/__tests__/terminated-test.cat-be.spec.ts:138]: a few too many spaces between the comma and the 1000

File: src/functions/sendCandidateResults/application/service/__tests__/terminated-test.cat-be.spec.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
```
Comment (rossfauldsbjss) [inline src/functions/sendCandidateResults/application/service/__tests__/terminated-test.cat-be.spec.ts:170]: spaces.

File: src/functions/sendCandidateResults/framework/__mocks__/test-data.cat-be.mock.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+          code: '4',
+          description: '',
+          outcome: 'DF',
+        },
+        {
+          code: '5',
+          description: '',
+          outcome: 'DF',
+        }],
+      showMeTellMeComments: '',
```
Comment (rossfauldsbjss) [inline src/functions/sendCandidateResults/framework/__mocks__/test-data.cat-be.mock.ts]: Should only have 3 show me questions, and should have two tell me questions.

File: src/functions/sendCandidateResults/framework/__tests__/request-scheduler.cat-be.spec.ts [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        outcomePayload: {
+          interface: NOTIFY_INTERFACE,
+          state: ProcessingStatus.ACCEPTED,
+          staff_number: '123456',
+          retry_count: 0,
+          error_message: null,
+        },
+      });
+      done();
+    },         1000);
```
Comment (rossfauldsbjss) [inline src/functions/sendCandidateResults/framework/__tests__/request-scheduler.cat-be.spec.ts:63]: spaces

File: src/functions/sendCandidateResults/framework/__tests__/request-scheduler.cat-be.spec.ts [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        outcomePayload: {
+          interface: NOTIFY_INTERFACE,
+          state: ProcessingStatus.FAILED,
+          staff_number: '123456',
+          retry_count: 3,
+          error_message: 'timed out',
+        },
+      });
+      done();
+    },         4500);
```
Comment (rossfauldsbjss) [inline src/functions/sendCandidateResults/framework/__tests__/request-scheduler.cat-be.spec.ts:132]: spaces

**Review by rossfauldsbjss — APPROVED** [review]
looks good.

File: src/functions/sendCandidateResults/application/service/__tests__/terminated-test.cat-be.spec.ts (hunk 2 above)
Comment (appleabs) [reply src/functions/sendCandidateResults/application/service/__tests__/terminated-test.cat-be.spec.ts:138]: without the spaces lint fails

**Review by Cullima — CHANGES_REQUESTED** [review]
Need to rebase with develop and update tests for ETA changes that went in yesterday.

**Review by Cullima — APPROVED** [review]

**Pull request merged**